### PR TITLE
Refactor ingress class templating

### LINKED ISSUE #173: Ingress Class Name Refactor
**Is your feature request related to a problem? Please describe.**

When setting up ingresses for a Helm deployment of Refinery with both a gRPC and  HTTP facing ingress I ran into unexpected templating that differed between the two. 

**Describe the solution you'd like**

Standardize and use default k8s Ingress object key 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by masonjlegere**
## Which problem is this PR solving?

- Closes #173

## Short description of the changes

Moves templating to be equal across the ingresses and updates the example default values file with an example.

### CODE REVIEW

File: charts/refinery/templates/ingress-grpc.yaml
Comment (pkanal) [inline charts/refinery/templates/ingress-grpc.yaml:27]: Overall I like the API change! We do have to make sure it is backwards compatible here in order for this not to be a breaking change so we should still check for the old value

```suggestion
  {{- if or .Values.grpcIngress.ingressClassName .Values.ingress.className }}
```

File: charts/refinery/templates/ingress-grpc.yaml
Comment (pkanal) [inline charts/refinery/templates/ingress-grpc.yaml:28]:
```suggestion
  ingressClassName: {{ or .Values.grpcIngress.ingressClassName .Values.ingress.className }}
```

File: charts/refinery/templates/ingress.yaml
Comment (pkanal) [inline charts/refinery/templates/ingress.yaml:27]:
```suggestion
  {{- if or .Values.ingress.ingressClassName .Values.ingressClassName }}
```

File: charts/refinery/templates/ingress.yaml
Comment (pkanal) [inline charts/refinery/templates/ingress.yaml:28]:
```suggestion
  ingressClassName: {{ or .Values.ingress.ingressClassName .Values.ingressClassName }}
```

File: charts/refinery/values.yaml
Comment (pkanal) [inline charts/refinery/values.yaml:341]: Could we add docs here for the deprecated keys as well and mention that they are deprecated?

**Review by pkanal — CHANGES_REQUESTED** [review]
This is a much clearer API to deal with 👍🏽 I have left some suggestions in order to not make this a breaking change before we move forward with it.

File: charts/refinery/values.yaml
Comment (masonjlegere) [reply charts/refinery/values.yaml:347]: Added some comments indicating that they're deprecated

**Comment by masonjlegere** [comment]
Sorry completely forgot I even opened this PR

**Review by MikeGoldsmith — DISMISSED** [review]
Thanks for the update @masonjlegere - looks good for me.

I'd like a second review by @TylerHelmuth as they have more HelmChart knowledge.

File: charts/refinery/templates/ingress.yaml
Comment (TylerHelmuth) [inline charts/refinery/templates/ingress.yaml:28]: Same here.

File: charts/refinery/templates/ingress-beta.yaml
Comment (TylerHelmuth) [inline charts/refinery/templates/ingress-beta.yaml:30]: This is not a valid use of helm's `if` flow control.  When I checkout this branch and run `helm template refinery ./charts/refinery --set ingressClassName=nginx --set ingress.enabled=true` I get helm template errors.  I think instead we want to use [helm's default function](https://helm.sh/docs/chart_template_guide/function_list/#default).

File: charts/refinery/templates/ingress-grpc.yaml
Comment (TylerHelmuth) [inline charts/refinery/templates/ingress-grpc.yaml:28]: Same here.

**Comment by kentquirk** [comment]
@masonjlegere are you able to make further changes to this one? We can't merge it as-is.

**Comment by masonjlegere** [comment]
Yes! Corrected and added a basic unit test for it but can remove the test if it's not wanted. I imagine they can get ran whenever `ct` is being used against `minikube` in CircleCI. This could also just be made into a proper integration test if the kubecontext gets set.

File: charts/refinery/test/ingress_template_test.go
Comment (TylerHelmuth) [reply charts/refinery/test/ingress_template_test.go:13]: Or if `ct` can't run this scenario, we'll go without it.

**Review by TylerHelmuth — COMMENTED** [review]
@masonjlegere please rebase/merge in main.  I'd like to see our new automated tests pass with this PR.

**Review by pkanal — DISMISSED** [review]

**Comment by masonjlegere** [comment]
I am not seeing the CircleCI test job as a status check 🤔

**Comment by TylerHelmuth** [comment]
@masonjlegere it looks like maybe you had a bad merge/rebase.  The files changed tab is showing a bunch of changes that are already in main.

**Pull request closed without merging**
